man/fi_profile.3.md

```diff
@@ -0,0 +1,222 @@
+
+*prof_fid*
+: Reference to a profile object associating with the target fid.
+
+*flags*
+: Flags to open and/or control a profiling.
+
+*context*
+: Context associated with the profiling operations.
+
+*varlist*
+: An array of profiling variables.
+
+*eventlist*
+: An array of profiling events.
+
+*count*
+: Number of variables or events in the array.
+
+*var_id*
+: An unqie ID for a variable.
+
+*event*
+: A unqie ID for an event. 
+
+*callback*
+: Callback function registered for an event.
+
+*param*
+: The parameter for the callback function.
+
+*size*
+: The size of the callback's param allocated when registering the callback.
+
+*data_size*
+: The amount of data that the provider is provided in the "param" 
+when the callback is called.
+
+*data*
+: An uint64_t value of a variable.
+
+# DESCRIPTION
+
+Fabric profiling interface allows users to obtain some of configuration or runtime properties from the providers in the form of "variables". Providers define variables and make them readable to users through the profiling interface. Providers define events and use them to report an operation occured or a state changed where one or more variables may change their values. Users can register callback functions to one or more events, and receive notifications from providers once an event occurs.
+
+In order to support the profiling interface, an endpoing or a domain in a provider needs to maintain a fid_profile object, and implement the profiling operations there. To starts a profiling,  an open call must be made on the target endpoint or domain object to obtain the fid_profile object. If the target object does not support profiling, the open call will fail with error code -FI_ENOSYS. Once a fid_profile object is obtained successfully, it can be used to query available variables or events, and read variables' values.
+
+# VARIABLES
+
+The profiling variables are defined in providers, and become accessable (readable) externally through the profiling interface.  A variable can be a configuration setting, a runtime state or a counter in an endpoint or a domain in a provider. Some variables are common and available in most providers, while others are provider-specific, and are only available in one provider. Users can query available variables and their values using the profiling interface.
+
+UNEXP_MSG_CNT is a common variable and it counts the number of unexpected message recevied and unmatched. The variable has a unint64_t data value.
+
+A variable is specified using struct fi_profile_var_desc.
+
+{% highlight c %}
+struct fi_profile_var_desc {
+	uint32_t id;
+	enum fi_datatype type;
+	size_t size;
+	const char *name;
+	const char *desc;
+};
+{% endhighlight %}
+
+## id
+
+Each variable has a unique identifier (id). Providers provide this id and users use it to get variable value.
+
+## type
+
+The data type of the variable. Most variables are in basic type, i.e, FI_INT, FI_UINT32, etc. FI_VOID is used for a struct data type. 
```

Comment:
See comment on first patch.  It'll be easier on the user to explicitly indicate the structure, when needed.